### DISCUSSION TITLE
Post static_mapping using REST API issue

### DISCUSSION THREAD

**Discussion by couraillon** [jaredhendrickson13/pfsense-api#583] [Q&A] [answered]
Hello!

Usually when I add a static mapping via the pfSense web interface, once the modification has been applied, the IP addresses are put in order. 

But when I add static mapping with the REST API using:
/api/v2/services/dhcp_server/static_mapping
I don't need to click on “Apply” because the changes have already been applied, which is fine, but the new static_mapping appears at the very bottom of the list, it is not ordered.

Do you have any advice for refreshing the order of static_mapping by IPs?

Thank you!

**Comment by jaredhendrickson13** [comment, marked as answer]
You can use [sorting](https://pfrest.org/QUERIES_FILTERS_AND_SORTING/#sorting) for this. In your POST request to /api/v2/services/dhcp_server/static_mapping, you can set the `sort_by` parameter to `ipaddr` to sort the static mappings by IP address. You can sort by any field(s) available to the endpoint.

Since the webConfigurator sorts static mappings by IP address by default, I can make that the default for this endpoint in the next patch as well.

**Reply by couraillon** [reply]
Great, thank you very much for this quick and perfectly working answer! :D

May I take this opportunity to ask if there is an endpoint to:

1. Get the id of a specific MAC address? I'm currently retrieving all the interface (parent_id) configuration and using the response, but it's a bit time-consuming to read the whole json and compare each mac address to get its id.

2. Check if a MAC address is already registered? In the same way, I currently check if the MAC address is present in all the interface (parent_id) configuration, but it takes a lot of time.

Thank you!

**Reply by couraillon** [reply]
Would you have an advice on this? :)

Thank you!

**Reply by jaredhendrickson13** [reply]
There's not a great way to do that currently. Ideally you'd be able use a [query](https://pfrest.org/QUERIES_FILTERS_AND_SORTING/#queries) to target static mappings with specific field values. But queries for nested objects aren't currently possible since the only way for read all nested objects is through the parent object. This was a limitation in the core framework in v2 initially, but changes made in v2.1.0 and v2.2.0 should make it feasible to implement endpoints that read all nested objects directly, which would also allow them to be queried. 

Feel free to [open a feature request](https://github.com/jaredhendrickson13/pfsense-api/issues/new/choose) for that.

**Reply by couraillon** [reply]
Thank you for your help, I've opened a feature request and will try what you mentionned above.
